cupy/linalg/util.py

```diff
@@ -0,0 +1,43 @@
+from numpy import linalg
+
+import cupy
+
+
+def _assert_cupy_array(*arrays):
+    for a in arrays:
+        if not isinstance(a, cupy.core.ndarray):
+            raise linalg.LinAlgError(
+                'cupy.linalg only supports cupy.core.ndarray')
+
+
+def _assert_rank2(*arrays):
+    for a in arrays:
+        if a.ndim != 2:
+            raise linalg.LinAlgError(
+                '{}-dimensional array given. Array must be '
+                'two-dimensional'.format(a.ndim))
+
+
+def _assert_nd_squareness(*arrays):
+    for a in arrays:
+        if max(a.shape[-2:]) != min(a.shape[-2:]):
+            raise linalg.LinAlgError(
+                'Last 2 dimensions of the array must be square')
+
+
+def _tril(x, k=0):
+    m, n = x.shape
+    u = cupy.arange(m).reshape(m, 1)
```

Comment:
I think it would be better, but we should submit this and an improved `_triu` into an another PR because we will call these two functions in various ways.